scripts/github_api

```diff
@@ -123,6 +157,8 @@ class PullRequest:
 
         pr_info = request('GET', endpoint)
 
+        self.reference = os.path.basename(pr_info['statuses_url'])
```

Comment:
* That is the reference to the PRs head commit, which afaik is how GitHub references the pull request. I just found a clearer way to access it: pr_info['head']['sha']
* The pull requests API endpoint only provides the link to the statuses API endpoint for that PR. I haven't created a variable in PullRequest with the status because 1) since the status API is not used only for pull requests I can have an uniform way of accessing it for any type of reference and 2) getting the status would force another API call to be made at every instantiation of PullRequest. By accessing the Status object directly we avoid the extra call when the user only wants PR information and not PR Status information.